tests/functional/aws-node-sdk/test/object/rangeTest.js

```diff
@@ -48,67 +54,62 @@ function checkRanges(range, bytes, cb) {
         assert.deepStrictEqual(res.ContentType, 'application/octet-stream');
         assert.deepStrictEqual(res.Metadata, {});
 
-        // Write a file using the buffer so getRangeExec can then check bytes
-        fs.writeFile(`hashedFile.${bytes}.${range}`, res.Body, err => {
-            if (err) {
-                return cb(err);
-            }
-            // If the getRangeExec program fails, then the range is incorrect
-            return execFile('./getRangeExec', ['--check', '--size', total,
-                '--offset', begin, `hashedFile.${bytes}.${range}`],
-                err => cb(err));
-        });
-    })
-    .catch(cb);
+        // Write a file using the buffer so getRangeExec can then check bytes.
+        // If the getRangeExec program fails, then the range is incorrect.
+        return writeFileAsync(`hashedFile.${bytes}.${range}`, res.Body)
+        .then(() => execFileAsync('./getRangeExec', ['--check', '--size', total,
+            '--offset', begin, `hashedFile.${bytes}.${range}`]));
+    });
 }
 
 // Create 5MB parts and upload them as parts of a MPU
-function uploadParts(bytes, uploadId, cb) {
+function uploadParts(bytes, uploadId) {
     const eTags = [];
     const name = `hashedFile.${bytes}`;
-    async.times(2, (n, next) => {
-        execFile('dd', [`if=${name}`, `of=${name}.mpuPart${n + 1}`,
-            'bs=5242880', `skip=${n}`, 'count=1'], err => {
-                if (err) {
-                    return next(err);
-                }
-                return s3.uploadPartAsync({
-                    Bucket: bucket,
-                    Key: key,
-                    PartNumber: n + 1, // A MPU part cannot be 0
-                    UploadId: uploadId,
-                    Body: fs.createReadStream(`${name}.mpuPart${n + 1}`),
-                })
-                .then(res => {
-                    eTags[n] = res.ETag;
-                })
-                .then(() => next())
-                .catch(next);
-            });
-    }, err => cb(err, eTags));
+
+    return new Promise((resolve, reject) =>
+        async.times(2, (n, next) => {
```

Comment:
As per our conversation, uploading parts can happen in parallel in fact. I have updated this function to use `Promise.map` (which has the added benefit of returning an array of the `ETags`) 😄 .